custom_libs/subliminal_patch/providers/subsro.py

```diff
@@ -0,0 +1,260 @@
+from subliminal_patch.subtitle import Subtitle, guess_matches
+from subliminal.video import Episode, Movie
+from subzero.language import Language
+from subliminal_patch.exceptions import APIThrottled, TooManyRequests
+
+logger = logging.getLogger(__name__)
+
+
+class SubsRoSubtitle(Subtitle):
+    """SubsRo Subtitle."""
+
+    provider_name = "subsro"
+    hash_verifiable = False
+
+    def __init__(
+        self,
+        language,
+        title,
+        download_link,
+        imdb_id,
+        is_episode=False,
+        episode_number=None,
+        year=None,
+        release_info=None,
+        season=None,
+    ):
+        super().__init__(language)
+        self.title = title
+        self.page_link = download_link
+        self.imdb_id = imdb_id
+        self.matches = None
+        self.is_episode = is_episode
+        self.episode_number = episode_number
+        self.year = year
+        self.release_info = self.releases = release_info
+        self.season = season
+
+    @property
+    def id(self):
+        logger.info("Getting ID for SubsRo subtitle: %s. ID: %s", self, self.page_link)
+        return self.page_link
+
+    def get_matches(self, video):
+        matches = set()
+
+        if video.year and self.year == video.year:
+            matches.add("year")
+
+        if isinstance(video, Movie):
+            # title
+            if video.title:
+                matches.add("title")
+
+            # imdb
+            if video.imdb_id and self.imdb_id == video.imdb_id:
+                matches.add("imdb_id")
+
+            # guess match others
+            matches |= guess_matches(
+                video,
+                guessit(
+                    f"{self.title} {self.season} {self.year} {self.release_info}",
+                    {"type": "movie"},
+                ),
+            )
+
+        else:
+            # title
+            if video.series:
+                matches.add("series")
+
+            # imdb
+            if video.series_imdb_id and self.imdb_id == video.series_imdb_id:
+                matches.add("imdb_id")
+
+            # season
+            if video.season == self.season:
+                matches.add("season")
+
+            # episode
+            if {"imdb_id", "season"}.issubset(matches):
+                matches.add("episode")
+
+            # guess match others
+            matches |= guess_matches(
+                video,
+                guessit(
+                    f"{self.title} {self.year} {self.release_info}", {"type": "episode"}
+                ),
+            )
+
+        self.matches = matches
+        return matches
+
+
+class SubsRoProvider(Provider, ProviderSubtitleArchiveMixin):
+    """SubsRo Provider."""
+
+    languages = {Language(lang) for lang in ["ron", "eng"]}
+    languages.update(set(Language.rebuild(lang, forced=True) for lang in languages))
+    languages.update(set(Language.rebuild(lang, hi=True) for lang in languages))
```

Comment:
Why rebuild each language for forced and HI if there's none on this provider?

Reply:
Wanted to cover as much options as possible, but I understand if you want me to remove it.

Reply:
If there's no forced or HI subtitles on this privider, there's no point in adding support for it 😄

Thanks!